src/httpserver.cpp

```diff
@@ -215,6 +215,20 @@ static void http_request_cb(struct evhttp_request* req, void* arg)
     {
         WITH_LOCK(g_requests_mutex, g_requests.insert(req));
         g_requests_cv.notify_all();
+
+        auto conn = evhttp_request_get_connection(req);
+
+        // Close callback to clear active but running request. This is also
+        // called if the connection is closed after a successful request, but
+        // complete callback set below already cleared the state.
+        evhttp_connection_set_closecb(conn, [](evhttp_connection* conn, void* arg) {
+            auto req = static_cast<evhttp_request*>(arg);
+            LOCK(g_requests_mutex);
+            auto n{g_requests.erase(req)};
+            if (n == 1 && g_requests.empty()) g_requests_cv.notify_all();
```

Comment:
I think because the only call to `g_requests_cv.wait(...)` has a predicate that waits for `g_requests.empty()`. I've removed the dependency in the latest patch